go/test/endtoend/backup/vtbackup/backup_only_test.go

```diff
@@ -0,0 +1,313 @@
+		assert.Nil(t, err)
+
+		if startTablet {
+			err = tablet.VttabletProcess.Setup()
+			assert.Nil(t, err)
+		}
+	}
+
+	if initShardMaster {
+		// choose master and start replication
+		err := localCluster.VtctlclientProcess.InitShardMaster(keyspaceName, shardName, cell, master.TabletUID)
+
+		if err != nil {
+			fmt.Println("*********______________****************")
+			fmt.Println(err)
+			fmt.Println(master.HTTPPort)
+			fmt.Println(replica1.HTTPPort)
+			time.Sleep(5 * time.Minute)
+		}
+		assert.Nil(t, err)
+	}
+}
+
+func restore(t *testing.T, tablet *cluster.Vttablet, tabletType string, waitForState string) {
+	// Erase mysql/tablet dir, then start tablet with restore enabled.
+
+	log.Info("restoring tablet %s", time.Now())
+	resetTabletDirectory(t, *tablet, true)
+
+	err := tablet.VttabletProcess.CreateDB(keyspaceName)
+	assert.Nil(t, err)
+
+	// Start tablets
+	tablet.VttabletProcess.ExtraArgs = []string{"-db-credentials-file", dbCredentialFile}
+	tablet.VttabletProcess.TabletType = tabletType
+	tablet.VttabletProcess.ServingStatus = waitForState
+	tablet.VttabletProcess.SupportsBackup = true
+	err = tablet.VttabletProcess.Setup()
+	assert.Nil(t, err)
+}
+
+func resetTabletDirectory(t *testing.T, tablet cluster.Vttablet, initMysql bool) {
+
+	extraArgs := []string{"-db-credentials-file", dbCredentialFile}
+	tablet.MysqlctlProcess.ExtraArgs = extraArgs
+
+	// Shutdown Mysql
+	err := tablet.MysqlctlProcess.Stop()
+	assert.Nil(t, err)
+	// Teardown Tablet
+	err = tablet.VttabletProcess.TearDown()
+	assert.Nil(t, err)
+
+	// Empty the dir
+	err = os.RemoveAll(tablet.VttabletProcess.Directory)
+	assert.Nil(t, err)
+
+	if initMysql {
+		// Init the Mysql
+		tablet.MysqlctlProcess.InitDBFile = newInitDBFile
+		err = tablet.MysqlctlProcess.Start()
+		assert.Nil(t, err)
+	}
+
+}
+
+func tearDown(t *testing.T, initMysql bool) {
+	for _, tablet := range []cluster.Vttablet{*master, *replica1, *replica2} {
+		//Tear down Tablet
+		//err := tablet.VttabletProcess.TearDown()
+		//assert.Nil(t, err)
+		err := localCluster.VtctlclientProcess.ExecuteCommand("DeleteTablet", "-allow_master", tablet.Alias)
+		assert.Nil(t, err)
+
+		resetTabletDirectory(t, tablet, initMysql)
+	}
+
+	//// reset replication
+	//promoteSlaveCommands := "STOP SLAVE; RESET SLAVE ALL; RESET MASTER;"
+	//disableSemiSyncCommands := "SET GLOBAL rpl_semi_sync_master_enabled = false; SET GLOBAL rpl_semi_sync_slave_enabled = false"
+	//for _, tablet := range []cluster.Vttablet{*master, *replica1, *replica2} {
+	//	_, err := tablet.VttabletProcess.QueryTablet(promoteSlaveCommands, keyspaceName, true)
+	//	assert.Nil(t, err)
+	//	_, err = tablet.VttabletProcess.QueryTablet(disableSemiSyncCommands, keyspaceName, true)
+	//	assert.Nil(t, err)
+	//
+	//	for _, db := range []string{"_vt", "vt_insert_test"} {
+	//		_, err = tablet.VttabletProcess.QueryTablet(fmt.Sprintf("drop database if exists %s", db), keyspaceName, true)
+	//		assert.Nil(t, err)
+	//	}
+	//}
+
```

Comment:
@deepthi I need your input on how to fix replication. 

In my current setup, I run two sets of tests. Both of them work fine individually but not in continuation. 

There is this TearDown code which ideally should cleanup the mysql process just like Python but it is no working for me. I might be missing a step here but I am not able to find that. I have copied all the piece from Python. 

It works fine if I comment the reset replication code and instead initialize the MySQL again.

Reply:
What is the error you get?

Reply:
No Error, Just the test fails due to replication @ line 199 `cluster.VerifyRowsInTablet(t, replica1, keyspaceName, 1)`

Reply:
so rows are missing / not getting replicated?

Reply:
That's right. None of the rows replicated in replica.

Reply:
I'm torn about this. On the one hand, the intent of this test is not to test whether resetReplication works. On the other hand, maybe there is a problem with resetReplication that we should be fixing. 
@enisoc do you have any thoughts?

Reply:
@deepthi @enisoc 
FYI, The same flow works in python tests. I have looked multiple times if they are doing anything different but I am not able to spot it. 

If you guys can spot what am I missing then it will be very helpful.

Reply:
for anyone else following, we were able to resolve the replication issue by ordering the code correctly.